### PR TITLE
feat: Adds attributes to SchemaField

### PR DESCRIPTION
**Pull request by chalmerlowe**
Adds several new attributes to `SchemaField` and adds to the existing test suite to account for the new attributes.

* foreign_type_definition
* rounding_mode

### CODE REVIEW

**Review by suzmue — APPROVED** [review]

File: google/cloud/bigquery/enums.py [hunk 1]
```diff
@@ -306,6 +306,8 @@ class SqlTypeNames(str, enum.Enum):
     DATETIME = "DATETIME"
     INTERVAL = "INTERVAL"  # NOTE: not available in legacy types
     RANGE = "RANGE"  # NOTE: not available in legacy types
+    FOREIGN = "FOREIGN"  # NOTE: type acts as a wrapper for data types
+    # not natively understood by BigQuery unless translated
```
[the remark this answers is not in the document]
Comment (chalmerlowe) [reply google/cloud/bigquery/enums.py:310]: Yes, done.

File: google/cloud/bigquery/enums.py [hunk 2]
```diff
@@ -306,6 +306,8 @@ class SqlTypeNames(str, enum.Enum):
     DATETIME = "DATETIME"
     INTERVAL = "INTERVAL"  # NOTE: not available in legacy types
     RANGE = "RANGE"  # NOTE: not available in legacy types
+    FOREIGN = "FOREIGN"  # NOTE: type acts as a wrapper for data types
```
[the remark this answers is not in the document]
Comment (chalmerlowe) [reply google/cloud/bigquery/enums.py:309]: I supplemented the existing test that walks through some of enums with this enum in particular.

File: google/cloud/bigquery/schema.py [hunk 3]
```diff
@@ -331,6 +344,40 @@ def range_element_type(self):
...
+        2.5 => 2
+        """
+        return self._properties.get("roundingMode")
+
+    @property
+    def foreign_type_definition(self):
+        """Definition of the foreign data type.
+
+        Only valid for top-level schema fields (not nested fields).
+        If the type is FOREIGN, this field is required.
```
[the remark this answers is not in the document]
Comment (chalmerlowe) [reply google/cloud/bigquery/schema.py:377]: There is now. 😺

File: google/cloud/bigquery/enums.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chalmerlowe) [reply google/cloud/bigquery/enums.py:321]: I moved both lines of comment above the object FOREIGN so that they both align together.

**Pull request merged**